registry/handlers/layer.go

```diff
@@ -72,3 +72,24 @@ func (lh *layerHandler) GetLayer(w http.ResponseWriter, r *http.Request) {
 
 	handler.ServeHTTP(w, r)
 }
+
+// DeleteLayer performs a soft delete of a layer blob
+func (lh *layerHandler) DeleteLayer(w http.ResponseWriter, r *http.Request) {
+	context.GetLogger(lh).Debug("DeleteLayer")
+
+	layerStore := lh.Repository.Layers()
+	err := layerStore.Delete(lh.Digest)
+	if err != nil {
+		switch err := err.(type) {
+		case distribution.ErrUnknownLayer:
+			w.WriteHeader(http.StatusNotFound)
```

Comment:
Just a thought: I've always coded a delete where the thing being deleted wasn't found, to return a success, because the end state is the desired state. This potentially protects against a badly written client that retries all errors and thinks the request failed.